Gem adapter doesn't accept generic publishers

After changing `Publisher` implementation of request body, gem adapter started to fail:
```
2020-08-07T09:45:08.6548929Z java.util.concurrent.CompletionException: org.jruby.exceptions.TypeError: (TypeError) failed to coerce com.artipie.http.TrafficMetricSlice$MetricsPublisher to io.reactivex.Flowable
2020-08-07T09:45:08.6549197Z java.util.concurrent.CompletionException: org.jruby.exceptions.TypeError: (TypeError) failed to coerce com.artipie.http.TrafficMetricSlice$MetricsPublisher to io.reactivex.Flowable
2020-08-07T09:45:08.6549418Z 	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
2020-08-07T09:45:08.6549600Z 	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(Unknown Source)
2020-08-07T09:45:08.6549795Z 	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(Unknown Source)
2020-08-07T09:45:08.6549981Z 	at java.base/java.util.concurrent.CompletableFuture.thenApply(Unknown Source)
2020-08-07T09:45:08.6550178Z 	at java.base/java.util.concurrent.CompletableFuture$MinimalStage.thenApply(Unknown Source)
2020-08-07T09:45:08.6550364Z 	at com.artipie.http.async.AsyncSlice.response(AsyncSlice.java:63)
2020-08-07T09:45:08.6550538Z 	at com.artipie.http.Slice$Wrap.response(Slice.java:81)
2020-08-07T09:45:08.6550710Z 	at com.artipie.http.async.AsyncSlice.lambda$response$0(AsyncSlice.java:63)
2020-08-07T09:45:08.6550894Z 	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
2020-08-07T09:45:08.6551082Z 	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
2020-08-07T09:45:08.6551266Z 	at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source)
2020-08-07T09:45:08.6551444Z 	at io.reactivex.internal.observers.ConsumerSingleObserver.onSuccess(ConsumerSingleObserver.java:62)
2020-08-07T09:45:08.6551664Z 	at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback$FlatMapSingleObserver.onSuccess(SingleFlatMap.java:111)
2020-08-07T09:45:08.6551879Z 	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
2020-08-07T09:45:08.6552065Z 	at io.reactivex.subjects.SingleSubject.onSuccess(SingleSubject.java:141)
2020-08-07T09:45:08.6552259Z 	at hu.akarnokd.rxjava2.interop.SingleInterop.lambda$fromFuture$2(SingleInterop.java:82)
2020-08-07T09:45:08.6552456Z 	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source)
2020-08-07T09:45:08.6552640Z 	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source)
2020-08-07T09:45:08.6552983Z 	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
2020-08-07T09:45:08.6553163Z 	at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source)
2020-08-07T09:45:08.6553361Z 	at io.reactivex.internal.observers.ConsumerSingleObserver.onSuccess(ConsumerSingleObserver.java:62)
2020-08-07T09:45:08.6553541Z 	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
2020-08-07T09:45:08.6553752Z 	at io.reactivex.internal.operators.single.SingleDoOnSuccess$DoOnSuccess.onSuccess(SingleDoOnSuccess.java:60)
2020-08-07T09:45:08.6553945Z 	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
2020-08-07T09:45:08.6554132Z 	at io.reactivex.internal.operators.single.SingleMap$MapSingleObserver.onSuccess(SingleMap.java:64)
2020-08-07T09:45:08.6554352Z 	at io.reactivex.internal.operators.flowable.FlowableReduceSeedSingle$ReduceSeedObserver.onComplete(FlowableReduceSeedSingle.java:110)
2020-08-07T09:45:08.6554548Z 	at com.artipie.MeasuredStorage$2$1.onComplete(MeasuredStorage.java:261)
2020-08-07T09:45:08.6554730Z 	at wtf.g4s8.rio.file.ReadSubscriberState.onComplete(ReadSubscriberState.java:76)
2020-08-07T09:45:08.6554911Z 	at wtf.g4s8.rio.file.ReadRequest$Next.process(ReadRequest.java:127)
2020-08-07T09:45:08.6555081Z 	at wtf.g4s8.rio.file.ReadTaskQueue.run(ReadTaskQueue.java:93)
2020-08-07T09:45:08.6556860Z 	at wtf.g4s8.rio.file.CloseChanOnError.run(CloseChanOnError.java:25)
2020-08-07T09:45:08.6557085Z 	at wtf.g4s8.rio.file.ErrorOnException.run(ErrorOnException.java:33)
2020-08-07T09:45:08.6557247Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
2020-08-07T09:45:08.6557430Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
2020-08-07T09:45:08.6557597Z 	at java.base/java.lang.Thread.run(Unknown Source)
2020-08-07T09:45:08.6557819Z Caused by: org.jruby.exceptions.TypeError: (TypeError) failed to coerce com.artipie.http.TrafficMetricSlice$MetricsPublisher to io.reactivex.Flowable
2020-08-07T09:45:08.6558004Z 	at RUBY.response(<script>:52)
```

--- a/src/main/ruby/SubmitGem.rb
+++ b/src/main/ruby/SubmitGem.rb
@@ -49,7 +49,7 @@ class SubmitGem
     rnd_path = File.join(@gems, rnd_gem)
     jrnd_path = Paths::get(@gems, rnd_gem)
     AsyncResponse.new(
-        RxFile.new(jrnd_path).save(body)
+        RxFile.new(jrnd_path).save(Flowable::fromPublisher(body))
             .and_then(Completable::from_action{
               spec = Gem::Package.new(rnd_path).spec
               real_path = File.join(@gems, "#{spec.name}-#{spec.version}.gem")

